internal/statestore/backfill_test.go

```diff
@@ -275,20 +286,18 @@ func TestDeleteBackfill(t *testing.T) {
 	for _, tc := range testCases {
 		tc := tc
 		t.Run(tc.description, func(t *testing.T) {
-			errActual := service.DeleteBackfill(ctx, tc.ticketID)
-			if tc.expectedCode == codes.OK {
-				require.NoError(t, errActual)
+			errActual := service.DeleteBackfill(ctx, tc.backfillID)
+			require.NoError(t, errActual)
 
-				if tc.ticketID != "" {
-					_, errGetTicket := service.GetTicket(ctx, tc.ticketID)
-					require.Error(t, errGetTicket)
-					require.Equal(t, codes.NotFound.String(), status.Convert(errGetTicket).Code().String())
-				}
-			} else {
-				require.Error(t, errActual)
-				require.Equal(t, tc.expectedCode.String(), status.Convert(errActual).Code().String())
-				require.Contains(t, status.Convert(errActual).Message(), tc.expectedMessage)
+			if tc.backfillID != "" {
+				_, errGetTicket := service.GetTicket(ctx, tc.backfillID)
+				require.Error(t, errGetTicket)
+				require.Equal(t, codes.NotFound.String(), status.Convert(errGetTicket).Code().String())
 			}
+
+			// test that Backfill also deleted from last acknowledged sorted set
+			_, err = redis.Int64(conn.Do("ZSCORE", backfillLastAckTime, tc.backfillID))
```

Comment:
there is no point to perform this check if you are testing the case with an empty backfill id. I think it's better to put it under _if tc.backfillID != ""_ statement.